pkg/operator/controller/machineconfig/mcp.go

```diff
@@ -172,6 +172,8 @@ func (r *MachineConfigReconciler) CheckNodeObservabilityMCPStatus(ctx context.Co
 		r.Log.Info("NodeObservabilityMachineConfig current condition "+
 			"does not require MCP status check",
 			"MCP", mcp.Name, "conditions", r.CtrlConfig.Status.Conditions)
+
+		r.Unlock()
```

Comment:
Current branch was updated with main, which had #49 changes, hence is listed here too.